### PR TITLE
feat(logs): Optimize log loading

### PR DESCRIPTION
**Pull request by oceanlvr**

### CODE REVIEW

File: src/utils/logger.ts [hunk 1]
```diff
@@ -73,3 +73,65 @@ export const quitAndRenameLogger = () => {
...
+  filename: string,
+  onDataHandler: (buffer: Buffer) => void,
+  onErrorHandler: (err: NodeJS.ErrnoException) => void,
+) => {
+  // Open the file for reading and appending
+  fs.open(filename, 'r', function (err, fd) {
+    if (err) {
+      onErrorHandler(err)
+    }
+    let buffer
```
Comment (ysfscream) [inline src/utils/logger.ts:93]: Maybe you should set a default value to buffer? At least set the type definition.

File: src/views/log/index.vue [hunk 2]
```diff
@@ -35,21 +34,46 @@ import path from 'path'
...
+  private async loadData(): Promise<void> {
     const logger = getlogger('logs', 'info')
-    logger.info('logs page init')
+    const LOG_DIR = getOrCreateLogDir()
+    const LOG_PATH = path.join(LOG_DIR, 'log')
+    logger.info(LOG_PATH)
+    fs.readFile(LOG_PATH, 'utf-8', (error, data) => {
+      if (error) {
+        const logger = getlogger('logs', 'error')
+        logger.error(error.toString())
```
Comment (ysfscream) [inline src/views/log/index.vue:45]: Set an error message.

```js
this.$message.error(error.toString())
```

File: src/views/log/index.vue [hunk 3]
```diff
@@ -35,21 +34,46 @@ import path from 'path'
...
+        return
+      } else {
```
Comment (ysfscream) [inline src/views/log/index.vue:47]: The above has already returned, no `else` needed.

File: src/utils/logger.ts [hunk 4]
```diff
@@ -73,3 +73,65 @@ export const quitAndRenameLogger = () => {
...
+ */
+export const watchFileAppender = (
+  filename: string,
+  onDataHandler: (buffer: Buffer) => void,
+  onErrorHandler: (err: NodeJS.ErrnoException) => void,
+) => {
+  // Open the file for reading and appending
+  fs.open(filename, 'r', function (err, fd) {
+    if (err) {
+      onErrorHandler(err)
```
Comment (ysfscream) [inline src/utils/logger.ts:91]: If an error occurs, does the following code still need to be executed? if not, please stop it, add a `return`.

**Review by 0721Betty — APPROVED** [review]

**Review by ysfscream — APPROVED** [review]

**Pull request merged**